app/controllers/my_controller.rb

```diff
@@ -225,6 +235,22 @@ def redirect_if_password_change_not_allowed_for(user)
     false
   end
 
+  def write_settings(redirect_to:)
+    if request.patch?
+      @user.attributes = permitted_params.user
+      @user.pref.attributes = params[:pref] || {}
+      @user.pref[:no_self_notified] = (params[:no_self_notified] == '1')
+      if @user.save
+        @user.pref.save
+        @user.notified_project_ids =
+          (@user.mail_notification == 'selected' ? params[:notified_project_ids] : [])
+        set_language_if_valid @user.language
+        flash[:notice] = l(:notice_account_updated)
+        redirect_to(action: "#{redirect_to}")
```

Comment:
`action: redirect.to_s` or simply `action: redirect_to` if you trust the caller...